pandas/core/generic.py

```diff
@@ -353,7 +353,7 @@ def _construct_axes_from_arguments(
         return axes, kwargs
 
     @classmethod
-    def _get_axis_number(cls, axis):
+    def _get_axis_number(cls, axis) -> int:
```

Comment:
`axis: Axis`?

Reply:
The goal of this PR is to type the disallow decorator. due to issues with types already added to functions decorated with this decorator the PR needs to fix this.

gradual typing allows to add types... gradually. adding Axis here requires more fixes so has been left untyped intentionally to keep the diff more atomic.

Reply:
makes sense, thanks